src/Compilers/Core/MSBuildTask/DotnetHost.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    class DotnetHost
+    {
+        public string ToolName { get; }
+
+        public string PathToTool { get; }
+
+        public string CommandLineArgs { get; }
+
+        public DotnetHost(string toolName, string pathToTool, string commandLineArgs)
+        {
+            ToolName = toolName;
+            PathToTool = pathToTool;
+            CommandLineArgs = commandLineArgs;
+        }
+
+        public DotnetHost(string toolNameWithoutExtension, string commandLineArgs)
+        {
+            string pathToTool;
+            string toolName;
+            if (IsCliHost(out string pathToDotnet))
+            {
+                toolName = $"{toolNameWithoutExtension}.dll";
+                pathToTool = Utilities.GenerateFullPathToTool(toolName);
+                if (pathToTool != null)
```

Comment:
Yes, that is intended that we put `null` into `PathToTool`. I wanted to keep existing behavior for `GeneratePathToTool` - the previous version is here https://github.com/khyperia/roslyn/blob/19c146e0bdfd1d61cd307b6e1071871c23e349ea/src/Compilers/Core/MSBuildTask/ManagedCompiler.cs#L387 (it returns null).